third_party/repositories.bzl

```diff
@@ -20,170 +27,193 @@ def namespaced_git_repository(name, **kwargs):
        **kwargs
    )
 
+
+def namespaced_build_file(libs):
+    return """
+package(default_visibility = ["//visibility:public"])
+load("@build_bazel_rules_swift//swift:swift.bzl", "swift_c_module",
+"swift_library")
+""" + "\n\n".join(libs)
+
+
+def namespaced_swift_c_library(name, srcs, hdrs, includes, module_map):
+    return """
+cc_library(
+  name = "{name}Lib",
+  srcs = glob([
+    {srcs}
+  ]),
+  hdrs = glob([
+    {hdrs}
+  ]),
+  includes = [
+    {includes}
+  ],
+  linkstatic = True
+)
+
+swift_c_module(
+  name = "{name}",
+  deps = [":{name}Lib"],
+  module_map = "{module_map}",
+)
+""".format(**dict(name=name,
+                srcs=",\n".join(['"%s"' % x for x in srcs]),
+                hdrs =",\n".join(['"%s"' % x for x in hdrs]),
+                includes=",\n".join(['"%s"' % x for x in includes]),
+                module_map=module_map))
+
+
+def namespaced_swift_library(name, srcs, deps=None, cc_libs=None, defines=None):
+    deps = [] if deps == None else deps
+    cc_libs = [] if cc_libs == None else cc_libs
+    defines = [] if defines == None else defines
+    return """
+swift_library(
+    name = "{name}",
+    srcs = glob([{srcs}]),
+    module_name = "{name}",
```

Comment:
This only namespaces deps - I was confused about how modules would work with this namespacing thing. I'd suggest just renaming the lib.

Reply:
We only need to namespace the repo and deps references

Reply:
Module names shouldn't be altered IMO since we've namespaced which version of the library you're compiling with now